.github/workflows/go.yml

```diff
@@ -209,7 +209,7 @@ jobs:
       - name: Setup Python
         uses: actions/setup-python@f677139bbe7f9c59b41e40162b753c062f5d49a3 # v5.2.0
         with:
-          python-version: 3.8
+          python-version: 3.9
```

Comment:
Can we bump this directly to something like 3.12? As far as I understand, this python version is only used for running archery, so it shouldn't matter much which version we use?

Reply:
I agree, I had the same thought, I will update accordingly. Another one that I was thinking is whether we want to change the default `.env` to be bumped too to something like 3.12

Reply:
How about using the latest Python by `3.x` for this? If we can use `3.x` for this, we don't need to bump version periodically.

See also: https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#using-the-python-version-input

Reply:
I've bumped to 3.12